cmd/clusterctl/client/cluster/workload_cluster.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	utilkubeconfig "sigs.k8s.io/cluster-api/util/kubeconfig"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkloadCluster has methods for fetching kubeconfig of workload cluster from management cluster.
+type WorkloadCluster interface {
+	//Get workload cluster kubeconfig
+	GetKubeconfig(workloadClusterName string, namespace string) (string, error)
+}
+
+// workloadCluster implements WorkloadCluster.
+type workloadCluster struct {
+	proxy Proxy
+}
+
+// newWorkloadCluster returns a workloadCluster.
+func newWorkloadCluster(proxy Proxy) *workloadCluster {
+	return &workloadCluster{
+		proxy: proxy,
+	}
+}
+
+func (p *workloadCluster) GetKubeconfig(workloadClusterName string, namespace string) (string, error) {
+	cs, err := p.proxy.NewClient()
+	if err != nil {
+		return "", err
+	}
+
+	obj := client.ObjectKey{
+		Namespace: namespace,
+		Name:      workloadClusterName,
+	}
+	dataBytes, err := utilkubeconfig.FromSecret(ctx, cs, obj)
+	return string(dataBytes), errors.Wrapf(err, "\"%s-kubeconfig\" not found in namespace %q", workloadClusterName, namespace)
```

Comment:
Uhmm, at first sight, this line gots me tricked
Might be better to explicitly handle err != nil instead of relying on:
- Wrapf returning nil in case error is nil 
- FromSecret returning nil in case error is not nil

Reply:
Also might be worth to consider if the err from FromSecret should be wrapped in order to provide a better UX